data/avro-schema/src/main/resources/avro/net/corda/data/p2p/markers/LinkManagerReplayMarker.avsc

```diff
@@ -0,0 +1,13 @@
+{
+  "type": "record",
+  "name": "LinkManagerReplayMarker",
+  "namespace": "net.corda.p2p.markers",
+  "fields": [
+    {
+      "name": "message",
```

Comment:
I think we can omit this field to save space. We include it in the sent marker, so that we can easily replay messages, but we don't include it in the other markers as they can all be linked together using the key which is the message identifier.

Reply:
I think we do need the `message`. I also think in you need to change the `MessageTracker` in the `DeliveryTracker` in `https://github.com/corda/corda-runtime-os/pull/1469` to process the a `LinkManagerReplayMarker` in the same way as a `LinkManagerSentMarker`.

The reason we need to do this, is that the markers topic is a compacted topic so we are only guaranteed to observe the last marker for a given message. Otherwise we will only be able to a replay messages once.

Reply:
The markers topic was never meant to be a compacted topic I believe. If it is now, we might have gone somewhere wrong. 

The markers topic was supposed to be a purely event stream and the corresponding markers state topic was supposed to be the compacted one.

Reply:
checking the latest setup that seems to be the case: https://github.com/corda/corda-runtime-os/blob/p2p-preview/applications/tools/p2p-test/p2p-setup/src/main/resources/p2p-kafka-setup-example.conf#L125-L153

Reply:
I think you are correct. We can probably just change the MessageTracker to update the `state` with the last `state` instead then, when a `LinkManagerReplayedMarker` is observed.

Reply:
yes, I think the message tracker can even remain agnostic of the replay markers, it only cares for sent & received/expired for the replay logic.